packages/contracts/contracts/DepositFeed.sol

```diff
@@ -0,0 +1,51 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+/**
+ * @title DepositFeed
+ */
+contract DepositFeed {
+    // Constant for address aliasing
+    uint160 private constant OFFSET = uint160(0x1111000000000000000000000000000000001111);
+
+    /**
+     * Event with the parameters required to derive transactions on L2.
+     */
+    event TransactionDeposited(
+        address indexed from,
+        address indexed to,
+        uint256 value,
+        uint256 gasLimit,
+        bool isCreation,
+        bytes data
+    );
+
+    /**
+     * Accepts deposits of ETH and data, and emits a TransactionDeposited event for use in deriving
+     * deposit transactions.
```

Comment:
The "ETH deposit" aspect of it is something that isn't really explained in the spec, is it?

This makes the deposit contract acts like a bridge for ETH (i.e. locking up the ETH) so that it can be minted on L2 (something we haven't specced either).

I have a few concerns with this:
1. This makes ETH deposits special instead of using the generic mechanism (i.e. use L1 to deposit in a locking contract, emit a minting transaction deposit to the deposit contract).
2. There is currently no way to reclaim ETH from this contract, which means that the ETH we'd mint on L2 would be "unbacked" (i.e. they'd be more OETH than ETH that can be reclaimed from our bridge).

Reply:
1. Well, ETH is special. But I take your point that we don't want the UX to be too different from other tokens. Ultimately I think we'll probably want to reuse the same bridge contracts we have now, but I do think we might want to. 
2. That's definitely not great, I agree, but I do think it's a necessary simplification right now as withdrawals would explode the design surface quite a bit.